docs/gc.md

```diff
@@ -110,11 +110,45 @@ A special mode of the `gencon` policy is known as *Concurrent Scavenge* (`-Xgc:c
+
+JNI critical is used when the programmer wants direct addressability of the object. To do that, JNI needs the object to be in a contiguous region of memory, or at least make it think the object is in a contiguous region. For that to happen, JNI critical creates a temporary array on the size of total elements in the original array. It then copies element by element to this temporary array. After the JNI critical is done it copies everything back, again element by element from the temporary array to the arraylet. This operation is quite expensive. Double mapping arraylets was introduced to solve these problems.
+
+#### Double Mapping
+The initial idea of double mapping was to make a large array, which is divided across the heap in different chunks, look like a contiguous array.
+
+For double mapping to happen we leverage the system's virtual memory. Virtual memory address space on 64-bit systems is very large, 64 bits worth in fact, compared to 32 bits in the 32-bit systems counterpart. However, physical memory is limited to a size much smaller in most cases. Using more virtual memory address space does not really cost the application anything, but using more physical memory would. Therefore, instead of using more physical memory and perform expensive copies, we map two different virtual memory addresses to the same physical memory address.
+
+![The diagram is explained in the surrounding text](./cr/arraylet_doublemap.png "Arraylet double mapping diagram")
+
+With double mapping, whenever JNI or JIT uses arraylets, they won’t need to create and/or copy element by element to a temporary array. The double mapping is created as soon as the array in the Java program is instantiated. Therefore, JNI and JIT only need to reference the contiguous memory double mapped mirror of the original array. By doing so, it saves a lot of time because the contiguous view of the array which is chucked in its arraylet leaves was already created and only needs to be referenced by JNI or JIT. Modifications made to the contiguous representation of the arraylet are reflected on the discontiguous one and vice-versa.
+
+Double mapping is enabled by default. If one wishes to disable it he must pass the command line option `-Xgc:disableArrayletDoubleMapping`.
+
+*Note:* Double mapping is only supported in Linux systems. Also, double mapping can not be supported if object heap happen to be allocated in Large (Huge) Pages. In such systems double mapping would be ignored even if requested explicitly. Please use [`-Xlp:objectheap:pagesize`](xlpobjectheap.md#-Xlp:objectheap) option to configure object heap in small pages (for example [`-Xlp:objectheap:pagesize=4k`](xlpobjectheap.md#-Xlp:objectheap) ). Please use [`-verbose:sizes`](xlpobjectheap.md#-Xlp:objectheap) for reference of page sizes available in the system.
```

Comment:
if object heap -> if the object heap
happen -> happens
would be -> will be
Please use -> Please use the
configure object heap -> configure the object heap
Please use -> Use
for reference of -> the see the

Reply:
The doc team will do a final sweep for style to ensure that the content is consistent with the rest of the docs. We'll probably merge and then tweak, as @bragaigor has put considerable effort in so far.